apps/messagegui/app.js

```diff
@@ -646,8 +646,18 @@ setTimeout(() => {
 }, 10); // if checkMessages wants to 'load', do that
 
 /* If the Bangle is unlocked by the user, treat that
-as a queue to stop repeated buzzing */
+as a queue to stop repeated buzzing.
+Also suspend the reload timeout while the watch is unlocked. */
+let isTimeoutSuspended;
 Bangle.on('lock',locked => {
-  if (!locked)
+  if (!locked) {
     require("messages").stopBuzz();
+    isTimeoutSuspended = !!unreadTimeout;
+    cancelReloadTimeout();
+  }
+  if (locked)
+    if (isTimeoutSuspended) {
```

Comment:
To avoid running more code than we need, we could do this - I'll leave it to you to decide though, perhaps the current way is more readable :)

```suggestion
  } else if (isTimeoutSuspended) {
```